sql/table.h

```diff
@@ -731,26 +733,30 @@ struct TABLE_SHARE
   uint16 row_start_field;
   uint16 row_end_field;
   uint32 hist_part_id;
-  List<void> free_parts;
-  bool free_parts_init;
+  Vers_field_stats** stat_trx;
+  ulonglong stat_serial;
```

Comment:
`LOCK_stat_serial` is **rwlock**, it protects not only `stat_serial` -- see its usage.